doc/sonic-application-extention/sonic-application-extention-hld.md

```diff
@@ -0,0 +1,1441 @@
+| /package/base-os-constraint | string          | no        | Base image version dependency constraint. Defaults to  '*': allows any version. |
+
+
+*base-os-constraint* should have the following format:
+
+```
+[>|>=|==|<|<=|^|!|!=]<version>
+```
+
+Example:
+
+```json
+{
+  "package": {
+    "base-os-constraint": ">1.0.0"
+  }
+}
+```
+
+
+*depends*, *conflicts* fields are defined to be in the following format:
+
+```
+<package-name>[>|>=|==|<|<=|^|!|!=]<version>,[>|>=|==|<|<=|^|!|!=]<version>,...
+```
+
+Examples:
+
+```json
+{
+  "package": {
+    "depends": "swss>=1.0.0,!=1.2.2,<=3.0.0"
+  }
+}
+```
+
+or
+
+```json
+{
+  "package": {
+    "breaks": "syncd^1.0.0"
+  }
+}
+```
+
+### SONiC Package Changelog
+
+<!-- omit in toc -->
+###### Manifest path
+
+| Path                                   | Type            | Mandatory | Description                             |
+| -------------------------------------- | --------------- | --------- | --------------------------------------- |
+| /package/changelog                     | dict            | no        | Changelog dictionary.                   |
+| /package/changelog/\<version\>         | dict            | yes       | Package version.                        |
+| /package/changelog/\<version\>/changes | list of strings | yes       | Changelog messages for a given version. |
+| /package/changelog/\<version\>/author  | string          | yes       | Author name.                            |
+| /package/changelog/\<version\>/email   | string          | yes       | Author's email address.                 |
+| /package/changelog/\<version\>/date    | string          | yes       | Date and time in RFC 2822 format.       |
+
+Example:
+
+```json
+{
+  "package": {
+    "changelog": {
+      "1.0.0": {
+        "changes": ["Initial release"],
+        "author": "Stepan Blyshchak",
+        "email": "stepanb@nvidia.com",
+        "date": "Mon, 25 May 2020 12:24:30 +0300"
+      },
+      "1.1.0": {
+        "changes": [
+          "Added functionality",
+          "Bug fixes"
+        ],
+        "author": "Stepan Blyshchak",
+        "email": "stepanb@nvidia.com",
+        "date": "Fri, 23 Oct 2020 12:26:08 +0300"
+      }
+    }
+  }
+}
+```
+
+This information will be useful for user, so a command to show changelog for a package is added:
+
+
+```
+admin@sonic:~$ sonic-package-manager show package changelog <some-package>
+1.0.0:
+
+  * Initial release
+
+-- Stepan Blyshchak <stepanb@nvidia.com>  Mon, 25 May 2020 12:24:30 +0300
+
+1.1.0
+
+  * Added functionality
+  * Buf fixes
+
+-- Stepan Blyshchak <stepanb@nvidia.com>  Fri, 23 Oct 2020 12:26:08 +0300
+
+```
+
+### SONiC Docker Container Resource Restrictions
+
+This feature will allow user to specify resource restrictions for a container via FEATURE table in CONFIG DB.
+This feature is not related to SONiC Application Extension Design, as can be applied to any existing SONiC
```

Comment:
How is this FEATURE table is populated? based on manifest file? is this applicable for 3rd party docker as well?

Reply:
This is out of scope of this document. Please see above replies